sway/input/seat.c

```diff
@@ -65,27 +65,34 @@ static void seat_container_destroy(struct sway_seat_container *seat_con) {
 	free(seat_con);
 }
 
-static void seat_send_focus(struct sway_seat *seat,
-		struct sway_container *con) {
-	if (con->type != C_VIEW) {
-		return;
-	}
-	struct sway_view *view = con->sway_view;
-	if (view->type == SWAY_VIEW_XWAYLAND) {
-		struct wlr_xwayland *xwayland =
-			seat->input->server->xwayland;
-		wlr_xwayland_set_seat(xwayland, seat->wlr_seat);
-	}
-	view_set_activated(view, true);
-	struct wlr_keyboard *keyboard =
-		wlr_seat_get_keyboard(seat->wlr_seat);
-	if (keyboard) {
-		wlr_seat_keyboard_notify_enter(seat->wlr_seat,
-				view->surface, keyboard->keycodes,
-				keyboard->num_keycodes, &keyboard->modifiers);
+static void seat_send_focus(struct sway_container *con, struct sway_seat *seat) {
+	if (con->type == C_VIEW) {
+		if (!seat_is_input_allowed(seat, con->sway_view->surface)) {
+			wlr_log(L_DEBUG, "Refusing to set focus, input is inhibited");
+			return;
+		}
+		struct sway_view *view = con->sway_view;
+		if (view->type == SWAY_VIEW_XWAYLAND) {
+			struct wlr_xwayland *xwayland =
+				seat->input->server->xwayland;
+			wlr_xwayland_set_seat(xwayland, seat->wlr_seat);
+		}
+		view_set_activated(view, true);
+		struct wlr_keyboard *keyboard =
+			wlr_seat_get_keyboard(seat->wlr_seat);
+		if (keyboard) {
+			wlr_seat_keyboard_notify_enter(seat->wlr_seat,
+					view->surface, keyboard->keycodes,
+					keyboard->num_keycodes, &keyboard->modifiers);
+		} else {
+			wlr_seat_keyboard_notify_enter(
+					seat->wlr_seat, view->surface, NULL, 0, NULL);
+		}
 	} else {
-		wlr_seat_keyboard_notify_enter(
-				seat->wlr_seat, view->surface, NULL, 0, NULL);
+		for (int i = 0; i < con->children->length; ++i) {
+			struct sway_container *child = con->children->items[i];
+			seat_send_focus(child, seat);
```

Comment:
Hmm, when focusing a container, do we want to call `wlr_seat_keyboard_notify_enter` on each view?

Reply:
I don't know. Do we? From the look of it, only one view can have keyboard focus so it ends up giving keyboard focus to the last view. People aren't likely to type stuff when using focus parent anyway.

Reply:
Yeah, I was wondering if it would make more sense to clear the keyboard focus.

Reply:
No, we don't want to call notify_enter.

Reply:
I've pushed a commit to address this, however this causes a usability issue where if you have a single view on the workspace and then run `focus parent` (eg. by mistake), there's no visible indication that the workspace itself is focused and anything you type has no effect.